package/kedro_viz/server.py

```diff
@@ -27,6 +34,29 @@ def is_localhost(host) -> bool:
     return host in ("127.0.0.1", "localhost", "0.0.0.0")
 
 
+def resolve_data_set_patterns(catalog: DataCatalog, pipelines: Dict[str, Pipeline]):
+    """Resolve data set patterns in the data catalog by matching them against
+    the data sets in the pipelines.
+    """
+
+    data_set_names = {
+        data_set_name
+        for pipeline in pipelines.values()
+        for data_set_name in pipeline.data_sets()
```

Comment:
ALso just flagging to the team that this method is renamed to `pipeline.datasets()` on the Kedro `develop` branch